### PR TITLE
add context layers for map report

### LINKED ISSUE #385: Map report should show context layers
screenshot 2018-10-12 at 06 57 24

Can we have the other context layers shown in the map:

• hazard
• aggregation

Having the OSM tiles as backdrop would also be nice but I suspect that is going to give us printing hassles in QGIS so skip that at the first sign of trouble. (see #376)

This was also an INGC request when @gubuntu visited

### PR DESCRIPTION
**Pull request by myarjunar**
fix #385 

selection_311

### CODE REVIEW

**Comment by gubuntu** [comment]
looking good

**Review by boney-bun — APPROVED** [review]
LGTM

**Comment by myarjunar** [comment]
@boney-bun ah I still can't merge this. I need some feedback from @lucernae because he also works on adding basemap to the map report. thanks for reviewing anyway!

File: tasks/analysis.py [hunk 1]
```diff
@@ -435,11 +436,29 @@ def process_impact_result(self, impact_result, analysis_id):
             # pass on template path according to headless service
             custom_template_path = headless_template_path
 
+        # define the layer order of the map report
+        layer_order = (
+            list(settings.LAYER_ORDER) if settings.LAYER_ORDER else None)
+        if layer_order:
+            layer_source = {
```
Comment (lucernae) [inline tasks/analysis.py:443]: This will miss the layer styles (because it's not in QML). But we will handle that later.

File: tasks/analysis.py [hunk 2]
```diff
@@ -435,11 +436,29 @@ def process_impact_result(self, impact_result, analysis_id):
...
+                'hazard': get_layer_path(analysis.hazard_layer),
+                'exposure': get_layer_path(analysis.exposure_layer)
+            }
+            if analysis.aggregation_layer:
+                layer_source.update({
+                    'aggregation': get_layer_path(analysis.aggregation_layer)})
+            else:
+                layer_order.remove('@aggregation')
+
+            layer_order = substitute_layer_order(layer_order, layer_source)
```
Comment (lucernae) [inline tasks/analysis.py:454]: Do we already include this function?

**Review by lucernae — APPROVED** [review]
LGTM!

File: tasks/analysis.py (hunk 2 above)
Comment (myarjunar) [reply tasks/analysis.py:454]: yes we did, it turns out that I copied the whole line of helpers file in realtime to geosafe :joy:

**Comment by lucernae** [comment]
By the way, it's pretty similar with what I'm currently doing with the basemap. But I can't merge my PR at the moment (needs some fixes on headless + inasafe). So you can go ahead on this one @myarjunar

File: tasks/analysis.py (hunk 1 above)
Comment (myarjunar) [reply tasks/analysis.py:443]: I took care of it in InaSAFE side IIRC. Let's see if it works.

File: tasks/analysis.py (hunk 2 above)
Comment (lucernae) [reply tasks/analysis.py:454]: Lol, I was doing the same thing with this so I added the functions on utils twice!

**Pull request merged**